src/ImageSharp/MetaData/ImageMetaData.cs

```diff
@@ -125,10 +132,46 @@ public double VerticalResolution
         public IList<ImageProperty> Properties { get; } = new List<ImageProperty>();
 
         /// <summary>
-        /// Gets or sets the number of times any animation is repeated.
-        /// <remarks>0 means to repeat indefinitely.</remarks>
+        /// Adds or updates the specified key and value to the <see cref="ImageMetaData"/>.
+        /// </summary>
+        /// <param name="key">The key of the metadata to add.</param>
+        /// <param name="value">The value of the element to add.</param>
+        /// <exception cref="ArgumentNullException">key is null.</exception>
+        /// <exception cref="ArgumentNullException">value is null.</exception>
+        /// <exception cref="ArgumentException">An element with the same key already exists in the <see cref="ImageMetaData"/>.</exception>
+        public void AddOrUpdateFormatMetaData(IImageFormat key, IImageFormatMetaData value)
```

Comment:
How about this pattern?
https://gist.github.com/antonfirsov/0eaf6ac9438655ea3affbf3212b33cc0

(Also posted on twitter)